### ISSUE TITLE
[Bug] fa2 paged prefill: non-causal sliding window is wrong on the first query tile when qo_len < kv_len

### ISSUE THREAD

**Issue by SamMausberg** [flashinfer-ai/flashinfer#4972]
### Before submitting

- [x] I have searched existing issues and did not find a solution or an existing report.

### Environment

GH200 (SM90), CUDA 12.8, torch 2.11.0+cu128, FlashInfer main at f67bc2ed (0.6.18).

### Bug description

`BatchPrefillWithPagedKVCacheWrapper` with `backend="fa2"`, `causal=False` and a finite `window_left` returns wrong outputs for the first 128 query rows of a request whose `qo_len` is shorter than its `kv_len`. Rows 128 and later, the causal case, and the fa3 backend all match an fp32 reference that masks `kv_pos < q_pos - window_left` with `q_pos = q_idx + kv_len - qo_len`, which is also what `mask.cuh`/`variants.cuh` implement for causal. The affected rows attend to (part of) the tokens left of their window.

The pattern (exactly one query tile, all heads, all head dims) points at the first-tile scheduling of the sliding window for non-causal requests in the FA2 paged prefill path rather than at the mask itself.

### Minimal reproduction

```python
import math, torch, flashinfer

torch.manual_seed(0)
qo_len, kv_len, H, HKV, D, page_size, window_left = 300, 4096, 8, 2, 128, 16, 333
num_pages = (kv_len + page_size - 1) // page_size
k = torch.randn(num_pages, page_size, HKV, D, dtype=torch.half, device="cuda")
v = torch.randn_like(k)
q = torch.randn(qo_len, H, D, dtype=torch.half, device="cuda")
qo_indptr = torch.tensor([0, qo_len]).int()
kv_indptr = torch.tensor([0, num_pages]).int()
kv_indices = torch.arange(num_pages).int()
last_page_len = torch.tensor([kv_len - (num_pages - 1) * page_size]).int()
workspace = torch.empty(128 << 20, dtype=torch.uint8, device="cuda")

outs = {}
for backend in ["fa2", "fa3"]:
    w = flashinfer.BatchPrefillWithPagedKVCacheWrapper(workspace, "NHD", backend=backend)
    w.plan(qo_indptr, kv_indptr, kv_indices, last_page_len, H, HKV, D, page_size,
           causal=False, window_left=window_left, q_data_type=torch.half)
    outs[backend] = w.run(q, (k, v)).float()

K = k.reshape(-1, HKV, D)[:kv_len].float().repeat_interleave(H // HKV, 1)
V = v.reshape(-1, HKV, D)[:kv_len].float().repeat_interleave(H // HKV, 1)
s = torch.einsum("qhd,khd->hqk", q.float(), K) / math.sqrt(D)
q_pos = torch.arange(qo_len, device="cuda")[:, None] + kv_len - qo_len
k_pos = torch.arange(kv_len, device="cuda")[None, :]
s = s.masked_fill(~(k_pos >= q_pos - window_left)[None], float("-inf"))
ref = torch.einsum("hqk,khd->qhd", torch.softmax(s, -1), V)
for backend, out in outs.items():
    err = (out - ref).abs().amax(dim=(1, 2))
    bad = (err > 1e-2).nonzero().flatten()
    print(backend, f"max err {err.max():.4f}", "bad rows", bad.numel(),
          f"({bad.min().item()}..{bad.max().item()})" if bad.numel() else "")
```

Output:

```
fa2 max err 0.2193 bad rows 128 (0..127)
fa3 max err 0.0002 bad rows 0
```

### Additional context

Found while extending the SM90 paged prefill tests (#4971); that test skips the fa2 cross-check for non-causal sliding-window cases because of this.

**Comment by SamMausberg** [comment]
Update: the equal-length case is affected as well, not only `qo_len < kv_len`. With `qo_len = kv_len = 1024`, `causal=False`, `window_left=333` (8 query heads, 2 KV heads, head_dim 128, bf16), the fa2 ragged prefill output is off by up to 0.58 against an fp32 reference (0.55 with `logits_soft_cap=30`), while fa3 matches to 0.002. Same with a paged KV cache and with an fp8 KV cache. `window_left=-1` is correct for both backends.
